packages/flutter/lib/src/services/system_chrome.dart

```diff
@@ -400,19 +400,19 @@ abstract final class SystemChrome {
   ///   @override
   ///   void didChangeDependencies() {
   ///     super.didChangeDependencies();
-  ///     _display = View.maybeOf(context)?.display;
+  ///     _view = View.maybeOf(context);
   ///   }
   ///
   ///   @override
   ///   void dispose() {
   ///     WidgetsBinding.instance.removeObserver(this);
-  ///     _display = null;
+  ///     _view = null;
   ///     super.dispose();
   ///   }
   ///
   ///   @override
   ///   void didChangeMetrics() {
-  ///     final ui.Display? display = _display;
+  ///     final ui.Display? display = _view?.display;
```

Comment:
If you cached the `display` property here, you won't actaully get the new size values of the display - you'll get the size values of whatever the display was when you cached it.